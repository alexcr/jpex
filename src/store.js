import React from 'react';
import { useLocalStore } from 'mobx-react-lite';
import { createStore } from './createStore';

const storeContext = React.createContext(null);

export const StoreProvider = ({children}) => {
  const store = useLocalStore(createStore);
  return (
    <storeContext.Provider value={store}>
      {children}
    </storeContext.Provider>
  );
}

export const useStore = () => {
  const store = React.useContext(storeContext);
  if (!store) {
    throw new Error('useStore must be used within a StoreProvider.');
  }
  return store;
}
