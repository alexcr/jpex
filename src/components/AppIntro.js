import React from 'react';
import styled from 'styled-components';
import { useStore } from '../store';
import { useUpload } from '../upload';

const Button = styled.span`
  border: none;
  cursor: pointer;
  padding: 0px;
  color: #006616;

  &:hover {
    color: #01b227;
  }
`;

export function AppIntro() {
  const store = useStore();
  const {open, getInputProps} = useUpload(store.setJSON);
  return (
    <>
      <input {...getInputProps()} />
      <p>
        Drag a JSON file to the area below or <Button onClick={open}>{'{ click here }'}</Button> to select one.
      </p>
    </>
  );
}
