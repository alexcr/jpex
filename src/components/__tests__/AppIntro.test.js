import React from 'react';
import { render } from 'enzyme';
import { AppIntro } from '../AppIntro';
import { StoreProvider } from '../../store';

describe('AppIntro', () => {
  it('should render intro paragraph correctly', () => {
    expect(
      render(<StoreProvider><AppIntro /></StoreProvider>)
    ).toMatchSnapshot();
  });
});
