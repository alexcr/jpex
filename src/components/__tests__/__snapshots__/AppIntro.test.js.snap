// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIntro should render intro paragraph correctly 1`] = `
Array [
  <input
    accept="application/json"
    autocomplete="off"
    style="display:none"
    tabindex="-1"
    type="file"
  />,
  <p>
    Drag a JSON file to the area below or 
    <span
      class="sc-bdnylx WjWW"
    >
      { click here }
    </span>
     to select one.
  </p>,
]
`;
